pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4795,40 +4794,60 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if is_period is False and name in c_OFFSET_DEPR_FREQSTR:
+                if is_period is False and name.upper() in c_OFFSET_DEPR_FREQSTR:
```

Comment:
also, little nitpick, but `if not is_period` is generally preferred over `if is_period is False` (and similarly for the `True` counterparts)

Reply:
thanks, `if not is_period` looks much better indeed. I replaced `if is_period is False` with it (and `if is_period is True` with  `if is_period`).